lib/internal/abort_controller.js

```diff
@@ -296,7 +297,9 @@ class AbortSignal extends EventTarget {
       } else if (!signal[kSourceSignals]) {
         continue;
       } else {
-        for (const sourceSignalWeakRef of signal[kSourceSignals]) {
+        const sourceSignalWeakRefs = ArrayFrom(signal[kSourceSignals]);
+        for (let i = 0; i < sourceSignalWeakRefs.length; i++) {
+          const sourceSignalWeakRef = sourceSignalWeakRefs[i];
```

Comment:
That's almost certainly less efficient, `ArrayFrom` iterate over the object, so we essentially double the number of iteration